maestro-app/src/main/java/bio/overture/maestro/app/infra/adapter/outbound/indexing/elasticsearch/AnalysisCentricElasticSearchAdapter.java

```diff
@@ -0,0 +1,223 @@
+package bio.overture.maestro.app.infra.adapter.outbound.indexing.elasticsearch;
+
+import static bio.overture.maestro.app.infra.adapter.outbound.indexing.elasticsearch.SearchAdapterHelper.*;
+import static bio.overture.maestro.domain.utility.StringUtilities.inputStreamToString;
+import static java.lang.String.format;
+import static java.util.Collections.singletonMap;
+
+import bio.overture.maestro.app.infra.config.RootConfiguration;
+import bio.overture.maestro.app.infra.config.properties.ApplicationProperties;
+import bio.overture.maestro.domain.api.message.IndexResult;
+import bio.overture.maestro.domain.entities.indexing.analysis.AnalysisCentricDocument;
+import bio.overture.maestro.domain.port.outbound.indexing.AnalysisCentricIndexAdapter;
+import bio.overture.maestro.domain.port.outbound.indexing.BatchIndexAnalysisCommand;
+import bio.overture.maestro.domain.utility.Parallel;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.fasterxml.jackson.databind.PropertyNamingStrategy;
+import io.github.resilience4j.retry.Retry;
+import io.vavr.control.Try;
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import lombok.NonNull;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.elasticsearch.action.admin.indices.alias.Alias;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.update.UpdateRequest;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.client.indices.CreateIndexRequest;
+import org.elasticsearch.client.indices.GetIndexRequest;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.core.io.Resource;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.retry.annotation.Retryable;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+@Slf4j
+public class AnalysisCentricElasticSearchAdapter implements AnalysisCentricIndexAdapter {
+
+  private RestHighLevelClient elasticsearchRestClient;
+
+  private final ObjectWriter analysisCentricJSONWriter;
+
+  private SnakeCaseJacksonSearchResultMapper searchResultMapper;
+
+  private final Resource analysisCentricIndex;
+
+  private final int documentsPerBulkRequest;
+
+  private final int maxRetriesAttempts;
+
+  private final String alias;
+  private final String indexName;
+  private final boolean enabled;
+  private static final int FALLBACK_MAX_RETRY_ATTEMPTS = 0;
+  private static final int FALL_BACK_WAIT_DURATION = 100;
+  private final long retriesWaitDuration;
+
+  @Inject
+  public AnalysisCentricElasticSearchAdapter(
```

Comment:
I think there is more room for refactoring here I have put something together but didn't test it, I can share it with you to check it first

Reply:
check this branch : refactor-adapters